padinfo/view/pantheon.py

```diff
@@ -56,37 +62,106 @@ async def deserialize(cls, dgcog, user_config: UserConfig, ims: dict):
+
+        # if monster has only mat types, show only mats, otherwise show everything else
+        type_list = []
+        if all(t.name in MAT_TYPES for t in monster.types):
+            series_name += ' (Mat)'
+            type_list = [m for m in base_list if all(t.name in MAT_TYPES for t in m.types)]
+        else:
+            type_list = [m for m in base_list if any(t.name not in MAT_TYPES for t in m.types)]
+        if len(type_list) > 0 and len(type_list) <= MAX_MONS_TO_SHOW:
+            return type_list, series_name, base_mon
+
+        filters = []
+
+        rarity_list = [m for m in type_list if m.rarity == base_mon.rarity]
+        filters.append('{}*'.format(base_mon.rarity))
+        if len(rarity_list) > 0 and len(rarity_list) <= MAX_MONS_TO_SHOW:
+            return rarity_list, cls.make_series_name(series_name, filters), base_mon
+
+        main_att = base_mon.attr1
+        sub_att = base_mon.attr2
+
+        # if the monster has nil main attribute, do comparisons using its subattribute
+        # if any monster in the list has nil main attribute, also compare using its subattribute
+        main_att_list = []
+        att_emoji = None
+        if main_att.name == NIL_ATT:
+            att_emoji = get_attribute_emoji_by_enum(sub_att)
+            main_att_list = [m for m in rarity_list if m.attr1.name == sub_att.name
+                                                       or (m.attr1.name == NIL_ATT
+                                                           and m.attr2.name == sub_att.name)]
+        else:
+            att_emoji = get_attribute_emoji_by_enum(main_att)
+            main_att_list = [m for m in rarity_list if m.attr1.name == main_att.name
+                                                       or (m.attr1.name == NIL_ATT
+                                                           and m.attr2.name == main_att.name)]
+        if len(main_att_list) > 0 and len(main_att_list) <= MAX_MONS_TO_SHOW:
+            # append after check this time, because if we go to subatt we only want one emoji
+            filters.append(att_emoji)
+            return main_att_list, cls.make_series_name(series_name, filters), base_mon
+
+        sub_att_list = [m for m in main_att_list if m.attr1.name == main_att.name
+                                                    and m.attr2.name == sub_att.name]
+        filters.append(get_attribute_emoji_by_monster(base_mon))
+        if len(sub_att_list) > 0 and len(sub_att_list) <= MAX_MONS_TO_SHOW:
+            return sub_att_list, cls.make_series_name(series_name, filters), base_mon
+
+        # if we've managed to get here, just cut it off
+        pantheon_list = sub_att_list[:MAX_MONS_TO_SHOW]
+        filters.append('(still too many, truncated)')
+
+        return pantheon_list, cls.make_series_name(series_name, filters), base_mon
 
 
 class PantheonView:
     VIEW_TYPE = 'Pantheon'
 
+    @staticmethod
+    def _pantheon_lines(monsters, base_monster):
+        if not len(monsters):
```

Comment:
Unnecessary check. If the list is empty, the comprehension will return an empty list.

Reply:
we cannot assume that a View came from the logic of a ViewState. This is good general safety

Reply:
also this is just good safety in general, why would you remove a safety check like this